Validation/EventGenerator/plugins/GenWeightValidation.cc

```diff
@@ -0,0 +1,177 @@
+#include "Validation/EventGenerator/interface/GenWeightValidation.h"
+
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "Validation/EventGenerator/interface/DQMHelper.h"
+#include "Validation/EventGenerator/interface/HepMCValidationHelper.h"
+using namespace edm;
+
+GenWeightValidation::GenWeightValidation(const edm::ParameterSet& iPSet) : wmanager_(iPSet, consumesCollector()) {
+  genParticleToken_ = consumes<reco::GenParticleCollection>(iPSet.getParameter<edm::InputTag>("genParticles"));
```

Comment:
All these parameters should better be added in the constructor' initialization list, and defined as `const` in the header whenever possible (I would say always, at a first glance)